src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -97,30 +97,31 @@ try
         case WM_MOUSEMOVE:
             if (WI_IsFlagSet(wParam, MK_LBUTTON))
             {
-                LOG_IF_FAILED(terminal->_MoveSelection(lParam));
+                LOG_IF_FAILED(publicTerminal->_MoveSelection(lParam));
                 return 0;
             }
             break;
         case WM_RBUTTONDOWN:
-            if (const auto& termCore{ terminal->_terminal }; termCore && termCore->IsSelectionActive())
+            if (publicTerminal->_terminal && publicTerminal->_terminal->IsSelectionActive())
             {
                 try
                 {
-                    const auto bufferData = termCore->RetrieveSelectedTextFromBuffer(false);
-                    LOG_IF_FAILED(terminal->_CopyTextToSystemClipboard(bufferData, true));
-                    TerminalClearSelection(terminal);
+                    const auto lock = publicTerminal->_terminal->LockForWriting();
+                    const auto bufferData = publicTerminal->_terminal->RetrieveSelectedTextFromBuffer(false);
+                    LOG_IF_FAILED(publicTerminal->_CopyTextToSystemClipboard(bufferData, true));
+                    publicTerminal->_ClearSelection();

```

Comment:
This adds the lock call and replaces `TerminalClearSelection` with `_ClearSelection` so that it isn't acquired again.